### PR TITLE
fix: use second-latest release tag for upgrade test on release branch

### LINKED ISSUE #585: `GetMostRecentReleaseTag` should check the branch name
When it's on a release branch, even the tag version is said to be a DevTagVersion, it should return the second latest release tag since the latest release hasn't been deployed yet

### PR DESCRIPTION
**Pull request by hh24k**
Closes #585 

Signed-off-by: Hai He <hai.he@enterprisedb.com>

### CODE REVIEW

File: .github/workflows/continuous-delivery.yml [hunk 1]
```diff
@@ -336,14 +337,21 @@ jobs:
           # use git describe to get the nearest tag and use that to build the version (e.g. 1.4.0+dev24 or 1.4.0)
           commit_version=$(git describe --tags --match 'v*' "${commit_sha}"| sed -e 's/^v//; s/-g[0-9a-f]\+$//; s/-\([0-9]\+\)$/+dev\1/')
           commit_short=$(git rev-parse --short "${commit_sha}")
-          git_version_tag=$(git describe --tags --match 'v*')
+          
+          # extract branch name
+          if [ ${{ github.event_name }} != 'pull_request' ]
```
Comment (litaocdl) [inline .github/workflows/continuous-delivery.yml:342]: should we add pull_request_target here, for pull_request_target we can get branch name from github_head_ref too

**Review by litaocdl — APPROVED** [review]

**Review by gbartolini — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Review by leonardoce — APPROVED** [review]

**Pull request merged**